doc/guides/env_vars.rst

```diff
@@ -0,0 +1,144 @@
+.. _env_vars:
+
+Environment Variables
+=====================
+
+Various pages in this documentation refer to setting Zephyr-specific
+environment variables. This page describes how.
+
+Setting Variables
+*****************
+
+Option 1: Just Once
+===================
+
+To set the environment variable :envvar:`MY_VARIABLE` to ``foo`` for the
+lifetime of your current terminal window:
+
+.. code-block:: console
+
+   # Linux and macOS
+   export MY_VARIABLE=foo
+
+   # Windows
+   set MY_VARIABLE=foo
+
+.. warning::
+
+  This is best for experimentation. If you close your terminal window, use
+  another terminal window or tab, restart your computer, etc., this setting
+  will be lost forever.
+
+  Using options 2 or 3 is recommended if you want to keep using the setting.
+
+Option 2: In all Terminals
+==========================
+
+On macOS and Linux, add the ``export MY_VARIABLE=foo`` line to your shell's
+startup script in your home directory. For Bash, this is usually
+:file:`~/.bashrc`.  The setting will then be available whenever you open a new
+terminal window.
+
+On Windows, you can install RapidEE, a freeware graphical environment variable
+editor, `using Chocolatey`_ in an Administrator command prompt:
+
+.. code-block:: console
+
+   choco install rapidee
+
+You can then run ``rapidee`` from your terminal to launch the program.
+
+Settings you make in RapidEE will be available whenever you open a new terminal
+window. They will also be available to other programs you run, including
+graphical ones.
+
+.. _env_vars_zephyrrc:
+
+Option 3: Using ``zephyrrc`` files
+==================================
+
+Choose this option if you don't want to make the variable's setting available
+to all of your terminals, but still want to save the value for loading into
+your environment when you are using Zephyr.
+
+**macOS and Linux:**
+
+Save the ``export MY_VARIABLE=foo`` line in the file :file:`~/.zephyrrc`,
+e.g. with this command:
+
+.. code-block:: console
+
+   cat >>~/.zephyrrc <<EOF
```

Comment:
I truly dislike here-documents, they are confusing for the user. Instead I would ask the user to create a file `~/.zephyrrc` with their favorite editor and then display the contents of it.